packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -70,28 +77,36 @@ export class BlobManager {
     }
 
     public async getBlob(blobId: string): Promise<IFluidHandle<ArrayBufferLike>> {
-        assert(this.blobIds.has(blobId) || this.pendingBlobIds.has(blobId), 0x11f /* "requesting unknown blobs" */);
-        return new BlobHandle(
-            `${BlobManager.basePath}/${blobId}`,
-            this.routeContext,
-            async () => this.getStorage().readBlob(blobId),
-        );
+        if (this.blobIds.has(blobId) || this.pendingBlobIds.has(blobId) || this.detachedBlobIds.has(blobId)) {
+            return new BlobHandle(
+                `${BlobManager.basePath}/${blobId}`,
+                this.routeContext,
+                async () => this.getStorage().readBlob(blobId),
+            );
+        }
+
+        const storageId = this.redirectTable?.get(blobId);
```

Comment:
I do not think there is big difference in order, so I'd move usage of this.redirectTable up and remove recursion in this function. I.e.

```typescript
let storageId = this.redirectTable?.get(blobId);
if (storageId  === undefined) {
   storageId = blobId;
}

assert(this.blobIds.has(blobId) || this.pendingBlobIds.has(blobId) || this.detachedBlobIds.has(blobId), 0x11f /* "requesting unknown blobs" */);
```

Also is there any way to reduce number of concepts? Is there a way to merge pendingBlobIds & detachedBlobIds (as they can't be used at the same time, and they have roughly same meaning - blobs that are in progress).

Reply:
I don't think it would make sense to merge these. detachedBlobIds could be removed since it's only used to validate the redirect table, which is probably not really necessary.